cmd/asset-syncer/postgresql_utils.go

```diff
@@ -156,16 +156,16 @@ func (m *postgresAssetManager) filesExist(chartFilesID, digest string) bool {
 	return err == nil && hasEntries
 }
 
-func (m *postgresAssetManager) insertFiles(chartFilesID string, files models.ChartFiles) error {
+func (m *postgresAssetManager) insertFiles(chartId string, files models.ChartFiles) error {
```

Comment:
why are you adding the chartID if it's already included in the files ID? (not sure if I am missing something)

Reply:
Do you mean that I could parse the files ID? Eg., in a current mongodb I see the following:
```
> db.charts.find({}, { _id: 1})
{ "_id" : "incubator/kafka" }
{ "_id" : "incubator/buzzfeed-sso" }
{ "_id" : "incubator/kubernetes-vault" }
...
> db.files.find({}, { _id: 1})
{ "_id" : "incubator/kubernetes-vault-0.2.1" }
{ "_id" : "incubator/buzzfeed-sso-0.1.2" }
{ "_id" : "incubator/neo4j-0.3.0" }
```
So yes, we could parse the chart name from there, but I don't want to depend on parsing or assumptions we have about what these will be (eg. `incubator/kubernetes-0-test-0.1.2`), so it seems sensible to just use the actual chartId verbatim?

Reply:
no, I meant that I don't know why adding the chartID is necessary now when we already have an ID (filesID) but since there are a bunch of changes I guess I am not seeing something.

Reply:
Ah - in the schema itself. The reason is that when I wrote the unit tests for Delete, I saw that currently we were only deleting charts and leaving the files dangling in the file table. To fix this, rather than manually deleting the files, I added an extra foreign key on the files table using the chartID and namespace with an `ON DELETE CASCADE` so that the DB takes care of this via the schema (and the test now passes - see ) https://github.com/kubeapps/kubeapps/blob/master/cmd/asset-syncer/postgresql_db_test.go#L192